### DISCUSSION TITLE
Render HTML based on component's prop with `nuxi generate`

### DISCUSSION THREAD

**Discussion by cossssmin** [nuxt/nuxt#29286] [Questions] [answered]
Can someone please help with a sanity check here as to why this doesn't work with `nuxi generate` and how to fix it?

```vue
<script setup>
  const props = defineProps({
    lang: {
      type: String,
      default: ''
    }
  })

  const localePath = (path) => {
    return props.lang ? `/${props.lang}${path}` : path
  }
</script>

<template>
  <NuxtLink
    :to="'/${lang}'"
    v-html="lang === 'en' ? 'Privacy Policy' : '...'"
  />
</template>
```

This doesn't show the "Privacy Policy" text when statically generating the site, I just get an empty `<a href="/"></a>` tag:

```vue
<template>
  <SiteFooter lang="en" />
</template>
```

**Comment by TheAlexLichter** [comment, marked as answer]
Using `v-html` on Vue components is generally not supported. Either render a `div` inside and us the directive on it, or use string interpolation (which is preferred for the simple example outlined here)!

**Reply by cossssmin** [reply]
Oh wow, didn’t know that. Was updating a Nuxt 2 project where it used to work. 

Thanks!
